pkg/controller/servingcert/secret_creating_controller.go

```diff
@@ -302,6 +302,19 @@ func (sc *ServiceServingCertController) syncService(key string) error {
 			}
 			return errors.New(serviceCopy.Annotations[ServingCertErrorAnnotation])
 		}
+		glog.V(4).Infof("updating already existing secret %s", secret.Name)
+		_, err = sc.secretClient.Secrets(serviceCopy.Namespace).Update(secret)
+		if err != nil {
+			// Use same annotation update logic as above.. Should be refactored
+			serviceCopy.Annotations[ServingCertErrorAnnotation] = fmt.Sprintf("secret/%v failed to update after regeneration: %v", secret.Name, err)
+			serviceCopy.Annotations[ServingCertErrorNumAnnotation] = strconv.Itoa(getNumFailures(serviceCopy) + 1)
```

Comment:
You can use `queue.NumRequeues` instead of annotations and then factor out the retries into where queuing is handled. Like https://github.com/openshift/machine-config-operator/blob/189a46306e429bb3aa3241670fe440d76a728ea7/pkg/operator/operator.go#L190-L209

Reply:
I can work on that as a followup to my generic controller logic in #52